docs/build.md

```diff
@@ -247,6 +249,33 @@ spec:
       name: icr-knbuild
 ```
 
+### Sources
+
+Represents remote artifacts, as in external entities that will be added to the build context before the actual build starts. Therefore, you may employ `.spec.sources` to download artifacts from external repositories.
```

Comment:
Although your first pass @otaviof is only supporting remote artifacts, we do know we have plans to expand that over time.  IMO we should go ahead and explain that here.

WDYT?

Reply:
Indeed, Gabe! I'll make sure the documentation reflects the plan to make `.spec.sources` an extension point by adding more "source types" later on.

Reply:
Please consider latest change.